### PR TITLE
feat: Provider Models (AuthProvider + concrete types)

### PR DESCRIPTION
**Pull request by demolaf**
- Precondition validations for auth provider configurations including Google email scope warnings, Facebook dependency checks, and phone number format validation
 - Provider-specific validation rules for email link settings, anonymous provider restrictions, and duplicate provider detection
 - Tests for all auth provider validation scenarios and configuration edge cases

### CODE REVIEW

File: auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt [hunk 1]
```diff
@@ -186,12 +230,42 @@ abstract class AuthProvider(open val providerId: String) {
...
+    }
 
     /**
      * Facebook Login provider configuration.
      */
     class Facebook(
+        /**
+         * The Facebook application ID.
+         */
+        val applicationId: String? = null,
```
Comment (demolaf) [inline auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt:267]: do we need the applicationId param?

File: auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt [hunk 2]
```diff
@@ -210,7 +284,31 @@ abstract class AuthProvider(open val providerId: String) {
...
+        fun validate(context: Context) {
+            if (!ProviderAvailability.IS_FACEBOOK_AVAILABLE) {
+                throw RuntimeException(
+                    "Facebook provider cannot be configured " +
+                            "without dependency. Did you forget to add " +
+                            "'com.facebook.android:facebook-login:VERSION' dependency?"
+                )
+            }
+
+            // Check application ID - either from parameter or string resources
```
Comment (demolaf) [inline auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt:297]: also this validation config from the old auth library if appId was missing

File: auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt [hunk 3]
```diff
@@ -186,12 +230,42 @@ abstract class AuthProvider(open val providerId: String) {
         providerId = Provider.GOOGLE.id,
         scopes = scopes,
         customParameters = customParameters
-    )
+    ) {
+        fun validate(context: Context) {
+            // TODO(demolaf): do we need this? since we are requesting this in AuthProvider.Google?
```
[the remark this answers is not in the document]
Comment (Lyokone) [reply auth/src/main/java/com/firebase/ui/auth/compose/configuration/AuthProvider.kt:235]: Was the validation done on the current version of the library?
